### PR TITLE
CAL-134: Mapping of CSEXRA attributes to ext.nitf.* fields.

### PR DESCRIPTION
**Pull request by troymohl**
#### What does this PR do?

CAL-134: Mapping for non-taxonomy CSEXRA NITF TRE fields.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?

@glenhein 
@bdeining
@peterhuffer
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).

@kcwire
@lessarderic
#### How should this be tested?

Run complete build with tests.
#### Any background context you want to provide?

The PR for CAL-134 will be broken out into multiple PRs.
#### What are the relevant tickets?

https://codice.atlassian.net/browse/CAL-134
This depends on: https://github.com/codice/alliance/pull/179
#### Checklist:
- [ ] Documentation Updated
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/CsexraAttribute.java [hunk 1]
```diff
@@ -85,20 +129,159 @@
...
     static final CsexraAttribute SNOW_DEPTH_MAX =
             new CsexraAttribute(Isr.SNOW_DEPTH_MAX_CENTIMETERS,
-                    SNOW_DEPTH_CAT,
+                    SNOW_DEPTH_CAT_NAME,
                     getSnowDepthAccessorFunction(Pair::getRight),
                     new IsrAttributes().getAttributeDescriptor(Isr.SNOW_DEPTH_MAX_CENTIMETERS),
                     "snowDepthMax",
                     ATTRIBUTE_NAME_PREFIX);
 
+    static final CsexraAttribute SNOW_DEPTH_CAT = new CsexraAttribute("snow-depth-category",
```
Comment (millerw8) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/CsexraAttribute.java]: Should these be constants in Isr?  Or anywhere?

File: catalog/imaging/imaging-transformer-nitf/src/test/java/org/codice/alliance/transformer/nitf/common/CsexraAttributeTest.java [hunk 2]
```diff
@@ -229,12 +229,840 @@ public void testSnowDepthMaxCategoryOther() throws NitfFormatException {
...
+        when(tre.getFieldValue(CsexraAttribute.SENSOR_NAME)).thenReturn("MS");
+
+        Serializable actual = CsexraAttribute.SENSOR.getAccessorFunction()
+                .apply(tre);
+
+        assertThat(actual, is(instanceOf(String.class)));
+        assertThat(actual, is("MS"));
+    }
+
+    @SuppressWarnings("unchecked")
```
Comment (lessarderic) [inline catalog/imaging/imaging-transformer-nitf/src/test/java/org/codice/alliance/transformer/nitf/common/CsexraAttributeTest.java]: Do we still need those?

File: catalog/imaging/imaging-transformer-nitf/src/test/java/org/codice/alliance/transformer/nitf/common/CsexraAttributeTest.java [hunk 3]
```diff
@@ -229,12 +229,840 @@ public void testSnowDepthMaxCategoryOther() throws NitfFormatException {
...
+        Serializable actual = CsexraAttribute.SENSOR.getAccessorFunction()
+                .apply(tre);
+
+        assertThat(actual, is(instanceOf(String.class)));
+        assertThat(actual, is("PAN"));
+
+    }
+
+    @Test
+    public void testSensorMS() throws NitfFormatException {
```
Comment (lessarderic) [inline catalog/imaging/imaging-transformer-nitf/src/test/java/org/codice/alliance/transformer/nitf/common/CsexraAttributeTest.java]: Nit: Should we name this `testSensorName` instead?

File: catalog/imaging/imaging-transformer-nitf/src/test/java/org/codice/alliance/transformer/nitf/common/CsexraAttributeTest.java [hunk 4]
```diff
@@ -229,12 +229,840 @@ public void testSnowDepthMaxCategoryOther() throws NitfFormatException {
...
     }
 
+    @Test
+    public void testSensorPan() throws NitfFormatException {
+        when(tre.getFieldValue(CsexraAttribute.SENSOR_NAME)).thenReturn("PAN");
+
+        Serializable actual = CsexraAttribute.SENSOR.getAccessorFunction()
+                .apply(tre);
+
+        assertThat(actual, is(instanceOf(String.class)));
```
Comment (lessarderic) [inline catalog/imaging/imaging-transformer-nitf/src/test/java/org/codice/alliance/transformer/nitf/common/CsexraAttributeTest.java]: Not sure we need to check that the instance if of type `String`. The next assert should fail if it isn't or if `actual` is `null`.

File: catalog/imaging/imaging-transformer-nitf/src/test/java/org/codice/alliance/transformer/nitf/common/CsexraAttributeTest.java [hunk 5]
```diff
@@ -229,12 +229,840 @@ public void testSnowDepthMaxCategoryOther() throws NitfFormatException {
...
+                NitfFormatException.class);
+
+        Serializable actual = CsexraAttribute.TIME_IMAGE_DURATION.getAccessorFunction()
+                .apply(tre);
+
+        assertThat(actual, nullValue());
+    }
+
+    @Test
+    public void testMaxGSDMin() throws NitfFormatException {
```
Comment (lessarderic) [inline catalog/imaging/imaging-transformer-nitf/src/test/java/org/codice/alliance/transformer/nitf/common/CsexraAttributeTest.java]: Nit: We usually camel-case acronyms, i.e., `testMaxGsdMin()`.

File: catalog/imaging/imaging-transformer-nitf/src/test/java/org/codice/alliance/transformer/nitf/common/CsexraAttributeTest.java [hunk 6]
```diff
@@ -229,12 +229,840 @@ public void testSnowDepthMaxCategoryOther() throws NitfFormatException {
...
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testSensorNotSet() throws NitfFormatException {
+        when(tre.getFieldValue(CsexraAttribute.SENSOR_NAME)).thenThrow(NitfFormatException.class);
+
+        Serializable actual = CsexraAttribute.SENSOR.getAccessorFunction()
+                .apply(tre);
+
+        assertThat(actual, nullValue());
```
Comment (lessarderic) [inline catalog/imaging/imaging-transformer-nitf/src/test/java/org/codice/alliance/transformer/nitf/common/CsexraAttributeTest.java]: Nit: Would look nicer if we did `is(nullValue())`.

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/CsexraAttribute.java [hunk 7]
```diff
@@ -85,20 +129,159 @@
...
+            SNOW_DEPTH_CAT_NAME,
+            tre -> TreUtility.convertToInteger(tre, SNOW_DEPTH_CAT_NAME),
```
Comment (troymohl) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/CsexraAttribute.java]: IDEA reports this as an illegal self-reference -- attempting to reference methods of a constructed object within the constructor.

File: catalog/imaging/imaging-transformer-nitf/src/test/java/org/codice/alliance/transformer/nitf/common/CsexraAttributeTest.java (hunk 3 above)
Comment (troymohl) [inline catalog/imaging/imaging-transformer-nitf/src/test/java/org/codice/alliance/transformer/nitf/common/CsexraAttributeTest.java]: This tests whether or not the Sensor attribute value is set to `MS` vs `PAN`.

Comment (lessarderic) [inline catalog/imaging/imaging-transformer-nitf/src/test/java/org/codice/alliance/transformer/nitf/common/CsexraAttributeTest.java]: Does it really matter from a unit test perspective, i.e., do we need both, or could we just keep the MS one and call it `testSensorNameIsMS()`?

File: catalog/imaging/imaging-transformer-nitf/src/test/java/org/codice/alliance/transformer/nitf/common/CsexraAttributeTest.java (hunk 5 above)
Comment (rzwiefel) [inline catalog/imaging/imaging-transformer-nitf/src/test/java/org/codice/alliance/transformer/nitf/common/CsexraAttributeTest.java]: And, Google code style guide also says thats the way you should do it. :)
https://google.github.io/styleguide/javaguide.html

Comment (troymohl) [inline catalog/imaging/imaging-transformer-nitf/src/test/java/org/codice/alliance/transformer/nitf/common/CsexraAttributeTest.java]: Fixed.

File: catalog/imaging/imaging-transformer-nitf/src/test/java/org/codice/alliance/transformer/nitf/common/CsexraAttributeTest.java (hunk 6 above)
Comment (troymohl) [inline catalog/imaging/imaging-transformer-nitf/src/test/java/org/codice/alliance/transformer/nitf/common/CsexraAttributeTest.java]: Updated tests to use is(nullValue())

File: catalog/imaging/imaging-transformer-nitf/src/test/java/org/codice/alliance/transformer/nitf/common/CsexraAttributeTest.java (hunk 4 above)
Comment (troymohl) [inline catalog/imaging/imaging-transformer-nitf/src/test/java/org/codice/alliance/transformer/nitf/common/CsexraAttributeTest.java]: Removed the instanceof checks.

**Pull request merged**